src/Microsoft.Extensions.OData.Migration/Formatters/Serialization/ODataMigrationOutputFormatter.cs

```diff
@@ -0,0 +1,385 @@
+﻿// ------------------------------------------------------------------------------
+// <copyright company="Microsoft Corporation">
+//     Copyright © Microsoft Corporation. All rights reserved.
+// </copyright>
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Extensions.OData.Migration.Formatters.Serialization
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Runtime.Serialization;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.AspNet.OData;
+    using Microsoft.AspNet.OData.Extensions;
+    using Microsoft.AspNet.OData.Formatter;
+    using Microsoft.AspNet.OData.Formatter.Serialization;
+    using Microsoft.AspNet.OData.Query;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Mvc.Formatters;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Primitives;
+    using Microsoft.Net.Http.Headers;
+    using Microsoft.OData;
+    using Microsoft.OData.Edm;
+    using Microsoft.OData.UriParser;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+    using ODataPath = AspNet.OData.Routing.ODataPath;
+
+    /// <summary>
+    /// Output formatter that supports V3 conventions in response bodies
+    /// </summary>
+    public class ODataMigrationOutputFormatter : ODataOutputFormatter
+    {
+        /// <summary>
+        /// Initialize the ODataMigrationOutputFormatter and specify that it only accepts JSON UTF8/Unicode input
+        /// </summary>
+        /// <param name="payloadKinds">The types of payloads accepted by this output formatter.</param>
+        public ODataMigrationOutputFormatter(IEnumerable<ODataPayloadKind> payloadKinds)
+            : base(payloadKinds)
+        {
+            SupportedMediaTypes.Add(MediaTypeHeaderValue.Parse("application/json"));
+            SupportedEncodings.Add(Encoding.UTF8);
+            SupportedEncodings.Add(Encoding.Unicode);
+        }
+
+        /// <summary>
+        /// Check for the presence of OData v3 headers
+        /// </summary>
+        /// <param name="context">Context that contains HTTP request</param>
+        /// <returns>True if specifically V3 OData request</returns>
+        public override bool CanWriteResult(OutputFormatterCanWriteContext context)
+        {
+            HttpRequest request = context.HttpContext.Request;
+            if (request.Headers.ContainsKey("dataserviceversion") ||
```

Comment:
I'm seeing these header name string constants duplicated in places.  They should be defined in one place and all code should use that.